tests/e2e/swarm_test.go

```diff
@@ -47,30 +50,133 @@ func (s *SwarmTestSuite) SetUpSuite(c *C) {
 	s.worker2 = inputparams.GetSwarmWorker2()
 	s.swarmNodes = inputparams.GetSwarmNodes()
 
-	s.volumeName = inputparams.GetUniqueVolumeName("swarm_test")
-	s.serviceName = inputparams.GetUniqueServiceName("swarm_test")
-
 	// Verify if swarm cluster is already initialized
 	out, err := dockercli.ListNodes(s.master)
 	c.Assert(err, IsNil, Commentf(out))
+}
+
+func (s *SwarmTestSuite) SetUpTest(c *C) {
+	s.volumeName = inputparams.GetUniqueVolumeName("swarm_test")
+	s.serviceName = inputparams.GetUniqueServiceName("swarm_test")
 
 	// Create the volume
-	out, err = dockercli.CreateVolume(s.master, s.volumeName)
+	out, err := dockercli.CreateVolume(s.master, s.volumeName)
 	c.Assert(err, IsNil, Commentf(out))
 
 	status := verification.VerifyDetachedStatus(s.volumeName, s.master, s.esxName)
 	c.Assert(status, Equals, true, Commentf("Volume %s is not detached", s.volumeName))
 }
 
-func (s *SwarmTestSuite) TearDownSuite(c *C) {
+func (s *SwarmTestSuite) TearDownTest(c *C) {
 	// Clean up the volume
 	out, err := dockercli.DeleteVolume(s.master, s.volumeName)
 	c.Assert(err, IsNil, Commentf(out))
 }
 
 var _ = Suite(&SwarmTestSuite{})
 
-// Test vDVS usage in swarm mode
+// Test vDVS usage during failover across different swarm nodes
+//
+// Test steps:
+// 1. Create a service with volume mounted
+// 2. Verify the service is up and running on one node
+// 3. Verify one container is spawned
+// 4. Verify the volume is in attached status
+// 5. Write data to the volume
+// 6. Shutdown the node on which the service is running
+// 7. Verify the service is restarted on a different node
+// 8. Verify the volume is in attached status
+// 9. Verify the data from this node
+// 10. Remove the service
+// 11. Verify the service is gone
+// 12. Verify the volume is in detached status
+func (s *SwarmTestSuite) TestFailoverAcrossSwarmNodes(c *C) {
+	misc.LogTestStart(c.TestName())
+
+	const (
+		testData = "Hello World!"
+		testFile = "hello.txt"
+		volPath  = "/vol"
+	)
+
+	fullVolumeName := verification.GetFullVolumeName(s.master, s.volumeName)
+	opts := "--mount type=volume,source=" + fullVolumeName + ",target=" + volPath + ",volume-driver=" + constant.VDVSPluginName + "--constraint node.role==worker --restart-condition on-failure" + constant.TestContainer
```

Comment:
Add some comments about the service to be created.  "Service will only be scheduled in the worker node and it will restart on failure".